### PR TITLE
Add crate for XPS operations

### PR DESCRIPTION
**Pull request by insipx**
Wanted to get some administrative tasks + skeleton out of the way before committing code

- add x86_64-unknown-linux-gnu to `rust-toolchain.toml`

I wasn't actually sure where this should live, but the way we discussed it at the offsite seemed like it would be in `libxmtp`.

this is the crate where we plan to impl `XmtpMlsClient` using [xps](https://github.com/xps-gateway)

~nothing will build yet until [reorg](https://github.com/xmtp/xps-gateway/pull/57) is merged into xps~

### CODE REVIEW

**Comment by jac18281828** [comment]
I agree this makes sense to me.  Thanks Andrew!

File: rust-toolchain [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [toolchain]
 channel = "stable"
 components = [ "rustc", "cargo", "clippy", "rustfmt", "rust-analyzer" ]
-targets = [ "wasm32-unknown-unknown" ]
+targets = [ "wasm32-unknown-unknown", "x86_64-unknown-linux-gnu"]
```
Comment (richardhuaaa) [inline rust-toolchain:4]: Noob question, why is this needed?

File: xps-client/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+homepage = "https://github.com/xmtp/libxmtp"
+repository = "https://github.com/xmtp/libxmtp"
+license = "MIT"
+keywords = ["ethereum", "did", "did:ethr", "jsonrpc"]
+categories = ["web-programming::websocket", "web-programming::http-server", "web-programming::http-client", "cryptography::cryptocurrencies"]
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+lib_xps = { git = "https://github.com/xmtp/xps-gateway", branch = "main", package = "lib-xps" }
```
Comment (richardhuaaa) [inline xps-client/Cargo.toml:16]: Noob question, I don't see `lib-xps` in that repo, what is this package responsible for?

Comment (insipx) [reply xps-client/Cargo.toml:16]: yeah it's part of the [reorg PR](https://github.com/xmtp/xps-gateway/pull/57), which splits the main.rs and lib.rs in the current `xps-gateway` into lib-xps and xps

File: rust-toolchain (hunk 1 above)
Comment (insipx) [reply rust-toolchain:4]: I saw that in dev/up the target was being added

This change isn't necessary, but would remove the need for that

Comment (richardhuaaa) [reply rust-toolchain:4]: Ah, looks like it's for the validation service, guess we're building it for a docker container on `linux/amd64`

**Review by richardhuaaa — APPROVED** [review]

File: xps-client/Cargo.toml (hunk 2 above)
Comment (insipx) [reply xps-client/Cargo.toml:16]: jsonrpsee generates a client and server from the defined api 

As it is defined right now, we don't have the client or server feature gated, which is something we have to do before a proper release, so it will pull in both

I'll add the feature gates to the [reorg](https://github.com/xmtp/xps-gateway/pull/57) before it goes in
